plugins/network-elements/nicira-nvp/src/main/java/com/cloud/network/guru/NiciraNvpGuestNetworkGuru.java

```diff
@@ -102,11 +112,11 @@ public NiciraNvpGuestNetworkGuru() {
     @Override
     protected boolean canHandle(final NetworkOffering offering, final NetworkType networkType, final PhysicalNetwork physicalNetwork) {
         // This guru handles only Guest Isolated network that supports Source nat service
-        if (networkType == NetworkType.Advanced && isMyTrafficType(offering.getTrafficType()) && offering.getGuestType() == Network.GuestType.Isolated
+        if (networkType == NetworkType.Advanced && isMyTrafficType(offering.getTrafficType())
+                && (offering.getGuestType() == Network.GuestType.Isolated || offering.getGuestType() == Network.GuestType.Shared)
```

Comment:
Is this check really needed? I mean `Isolated` and `Shared` are the [only two types of guest networks](https://github.com/apache/cloudstack/blob/95ae7963d5652595d62d6ace39de2eec9670413c/api/src/com/cloud/network/Network.java#L41).
If you would like to keep the check for sake of being future proof (that is, new guest types being added), then refactoring it to a method with a meaningful name (e.g. supportedGuestTypes(...)) would help readability.